proposals/dao/fip_33.ts

```diff
@@ -0,0 +1,218 @@
+const PCV_EQUITY_MINTER_APR_BPS = '1000'; // 10%
+
+/*
+
+TRIBE Buybacks
+
+DEPLOY ACTIONS:
+
+1. Deploy CR Oracle Keeper
+2. Deploy TRIBE/ETH Oracle
+3. Deploy TRIBE/USD Oracle
+4. Deploy TRIBE LBP Swapper
+5. Create TRIBE LBP pool
+6. Init TRIBE LBP Swapper
+7. Deploy PCV Equity Minter
+8. Deploy CR Oracle Guardian
+
+DAO ACTIONS:
+1. Make PCV Equity Minter a minter
+2. Make CR Oracle Keeper a minter
+3. Seed LBP Swapper with TRIBE
+4. Create ORACLE_ADMIN role
+5. Set ORACLE_ADMIN role to admin for CR Oracle
+6. Set ORACLE_ADMIN role to admin for CR Oracle Wrapper
+7. Grant Oracle Admin role to Collateralization Oracle Guardian
+8. TODO ORACLE admin to OA ??
+*/
+
+export const deploy: DeployUpgradeFunc = async (deployAddress, addresses, logging = false) => {
+  const {
+    core,
+    fei,
+    tribe,
+    feiEthPair,
+    weth,
+    chainlinkEthUsdOracleWrapper,
+    compositeOracle,
+    balancerLBPoolFactory,
+    collateralizationOracleWrapper,
+    chainlinkTribeEthOracle
+  } = addresses;
+
+  if (!core || !feiEthPair || !weth || !chainlinkEthUsdOracleWrapper || !compositeOracle) {
+    console.log(`core: ${core}`);
+    console.log(`feiEtiPair: ${feiEthPair}`);
+    console.log(`weth: ${weth}`);
+    console.log(`chainlinkEthUsdOracleWrapper: ${chainlinkEthUsdOracleWrapper}`);
+    console.log(`compositeOracle: ${compositeOracle}`);
+
+    throw new Error('An environment variable contract address is not set');
+  }
+
+  // 1.
+  const collateralizationOracleKeeperFactory = await ethers.getContractFactory('CollateralizationOracleKeeper');
+  const collateralizationOracleKeeper = await collateralizationOracleKeeperFactory.deploy(
+    core,
+    CR_KEEPER_INCENTIVE,
+    collateralizationOracleWrapper
+  );
+
+  logging && console.log('Collateralization Oracle Keeper: ', collateralizationOracleKeeper.address);
+
+  // 2.
+  const chainlinkTribeEthOracleWrapperFactory = await ethers.getContractFactory('ChainlinkOracleWrapper');
+  const chainlinkTribeEthOracleWrapper = await chainlinkTribeEthOracleWrapperFactory.deploy(
+    core,
+    chainlinkTribeEthOracle
+  );
+
+  logging && console.log('TRIBE/ETH Oracle Wrapper deployed to: ', chainlinkTribeEthOracleWrapper.address);
+
+  // 3.
+  const chainlinkTribeUsdCompositeOracleFactory = await ethers.getContractFactory('CompositeOracle');
+  const chainlinkTribeUsdCompositeOracle = await chainlinkTribeUsdCompositeOracleFactory.deploy(
+    core,
+    chainlinkTribeEthOracleWrapper.address,
+    chainlinkEthUsdOracleWrapper
+  );
+
+  logging && console.log('TRIBE/USD Composite Oracle deployed to: ', chainlinkTribeUsdCompositeOracle.address);
+
+  // 4.
+  const feiTribeLBPSwapperFactory = await ethers.getContractFactory('BalancerLBPSwapper');
+  const feiTribeLBPSwapper = await feiTribeLBPSwapperFactory.deploy(
+    core,
+    {
+      _oracle: chainlinkTribeUsdCompositeOracle.address,
+      _backupOracle: ethers.constants.AddressZero,
+      _invertOraclePrice: false, // TODO check this
```

Comment:
@Joeysantoro did you check this?

Reply:
https://github.com/fei-protocol/fei-protocol-core/pull/251